packages/datadog-instrumentations/src/ldapjs.js

```diff
@@ -0,0 +1,94 @@
+'use strict'
+
+const {
+  channel,
+  addHook,
+  AsyncResource
+} = require('./helpers/instrument')
+const shimmer = require('../../datadog-shimmer')
+
+function isString (value) {
+  return typeof value === 'string' || value instanceof String
+}
+
+function getCallbackArgIndex (args) {
+  let callbackIndex = -1
+  for (let i = args.length - 1; i >= 0; i--) {
+    if (typeof args[i] === 'function') {
+      callbackIndex = i
+      break
+    }
+  }
+  return callbackIndex
+}
+
+const emitters = new WeakMap()
+
+function wrapEmitter (corkedEmitter) {
+  shimmer.wrap(corkedEmitter, 'on', on => function (name, fn) {
+    if (typeof fn === 'function') {
+      const callbackResource = new AsyncResource('bound-anonymous-fn')
+      const bindedFn = callbackResource.bind(fn)
```

Comment:
I think that you shoud check before create the callback resource if it already exist in callbackMap. If someone does ```
res.on('error', fn)
res.on('error', fn)
```
we should add the same function twice.